src/ImageSharp/Formats/Gif/GifEncoderCore.cs

```diff
@@ -336,15 +341,19 @@ private void WriteColorTable<TPixel>(QuantizedFrame<TPixel> image, EndianBinaryW
             var rgb = default(Rgb24);
             using (IManagedByteBuffer colorTable = this.memoryManager.AllocateManagedByteBuffer(colorTableLength))
             {
-                Span<byte> colorTableSpan = colorTable.Span;
+                // TODO: Pixel operations?
+                ref TPixel paletteRef = ref MemoryMarshal.GetReference(image.Palette.AsSpan());
+                ref byte colorTableRef = ref MemoryMarshal.GetReference(colorTable.Span);
```

Comment:
By casting the `byte` reference to `Rgb24` reference, it's possible to have just one `Unsafe.Add()` inside the loop, also eliminating `i * 3`.

Reply:
Damn that's nice!